lib/haskell/natural4/src/LS/Rule.hs

```diff
@@ -477,6 +490,8 @@ data Expect = ExpRP      RelationalPredicate
             | ExpDeontic Rule -- regulative rule
             deriving (Eq, Ord, Show, Generic, Hashable, ToJSON)
 
+makePrisms ''Rule
```

Comment:
I had tried to `makePrisms ''Rule` in the past, but had got an error. I'm guessing there's some subtlety re where it should appear that I hadn't understood? At any rate, this is greatly appreciated!

Reply:
Yes, TH instructions break the ability of declarations to be mutually recursive. The place of a top-level TH splice basically splits the module into two parts, and stuff before cannot refer to stuff after. That's why the `makePrisms` call has to be that far away from the `Rule` data declaration.